build-scripts/package-checks.cake

```diff
@@ -0,0 +1,306 @@
+public class PackageChecker
+{
+    // The checks in this class basically do no more than what the programmer might 
+    // do in opening the package itself and examining the content. In particular, the
+    // packages are not actually installed, leaving that for manual testing before
+    // each release. This might be automated at some future point. However, they do
+    // provide some degree of regression testing, since changes in the package
+    // organization will result in a failure. 
+
+    static readonly string[] ENGINE_FILES = { "nunit.engine.dll", "nunit.engine.core.dll", "nunit.engine.api.dll", "testcentric.engine.metadata.dll" };
+    static readonly string[] AGENT_FILES = {
+        "nunit-agent.exe", "nunit-agent.exe.config", "nunit-agent-x86.exe", "nunit-agent-x86.exe.config", "nunit.engine.core.dll", "nunit.engine.api.dll", "testcentric.engine.metadata.dll" };
+    static readonly string[] CONSOLE_FILES = { "nunit3-console.exe", "nunit3-console.exe.config" };
+    static readonly string[] CONSOLE_FILES_NETCORE = { "nunit3-console.exe", "nunit3-console.dll", "nunit3-console.dll.config" };
+
+    private BuildParameters _parameters;
+    private ICakeContext _context;
+
+    public PackageChecker(BuildParameters parameters)
+    {
+        _parameters = parameters;
+        _context = parameters.Context;
+    }
+
+    public void CheckAllPackages()
+    {
+        string packageDir = _parameters.PackageDirectory;
+
+        bool isOK =
+            CheckNuGetPackage(
+                "NUnit.Console",
+                HasFile("LICENSE.txt")) &
```

Comment:
Just a personal opinion, but I think I would prefer separate variabels for each package and then check them all in the if below (I overlooked all the `&` at the first look at the file). Or perhaps just store all the results from all the `CheckNuGetPackage`s in an array/list, so we have a sequence of booleans, and then ensure that all the results are `true`.

Reply:
In fact, that's how my original code is structured. Each package type has a set of tests and a set of checks However, I didn't do that for nunit-console because only all packages have checks but only some have tests. I suppose the tests could be empty in that case. I can do another pass at this after my other stuff is merged.